compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2743,15 +2742,41 @@ impl<'a> Parser<'a> {
                                     ",".to_owned(),
                                     Applicability::MachineApplicable,
                                 );
+                                return Err(err);
                             }
-                            _ => {
-                                err.span_label(
-                                    arrow_span,
-                                    "while parsing the `match` arm starting here",
-                                );
+                        } else {
```

Comment:
Can't this logic be completely replaced with the `snapshot` parse attempt unconditionally? Or is it that we have advanced too much already and consumed part of the pattern by now specifically because of `&`?

Reply:
yeah, we would have consumed the `&Path` as the operator and RHS of operator `&` before getting here, which means we have to handle it differently.